### PR TITLE
feat(web/Spaces): Sidebar with Add To Space functionality

### PR DESCRIPTION
**Pull request by vmosharova**
> Finalize sidebar architecture with modular components,
> extract workspace header logic, add Safe-level navigation,
> expand test coverage to 438+ tests, improve UX with dialogs.

## What it solves
Implements the new Safe Sidebar.
Resolves: [WA-1842](https://linear.app/safe-global/issue/WA-1842/implement-add-to-space-button-for-non-spaces-safes)
Resolves: [WA-1478](https://linear.app/safe-global/issue/WA-1478/sidebar-for-the-safe-level)

## How this PR fixes it

**Architectural Changes:**
- Reorganized Sidebar components into focused subdirectories: `SidebarTopBar`, `SidebarSkeleton`, `SidebarCommonFooter`, `ApiCtaSidebar`, with index.ts exports following lazy-loading patterns
- Extracted `SafeSidebarWorkspaceHeader` to manage Safe-level workspace switching and navigation
- Moved `SafeSidebarContent` and `SafeSidebarVariant` into dedicated folders with improved separation of concerns

**New Components:**
- `SafeSidebarWorkspaceHeader` – Safe-level workspace header with space selector, back button, and transaction creation
- `BackToSpaceButton` – Navigate back to Space-level sidebar with guard against falsy spaceId
- `NewTransactionButton` – Quick transaction creation button in Safe sidebar
- `NavItem` – Refactored with improved styling and consistency
- `SpaceSelectorDropdown` – Significantly refactored with improved readability, added `useAddSafeToSpace` hook integration

**Features & Improvements:**
- Integrated Shadcn Dialog component for `AddToSpacePopupModal` wrapping
- Added `useAddSafeToSpace` hook with error handling and space limit validation
- Added Space limit enforcement (SAFE_ACCOUNTS_LIMIT) with user-friendly tooltips
- Hydration checks for Safe sidebar rendering
- Granular error notifications for space operations
- Enhanced Storybook stories with multiple Safe sidebar states

**Test Coverage:**
- Tests for: `SafeSidebarWorkspaceHeader`, `NewTransactionButton`, `SpaceSelectorDropdown`, `AddToSpacePopupModal`, `useAddSafeToSpace`
- Updated existing tests for `NavItem`, `SafeSidebarVariant`, etc.

## How to test it

Verify scenarios described on the ticket:
- Happy Path: Safe in Space + Deployed + user has Spaces
- Safe not in Space + Deployed + User Has Spaces
- Safe not in Space + Deployed + User Has NO Spaces
- Safe not in Space + Deployed + Safe Limit Reached

All unit tests should pass.

4. **Storybook Stories:**
   ```bash
   yarn workspace @safe-global/web storybook
   ```
   - Check `Features/Spaces/SafeSidebar` for multiple state variations
   - Verify all new components render correctly in stories

## Visual summary

```mermaid
graph TB
    subgraph "Before: Monolithic Sidebar"
        A1["Sidebar<br/>SpaceSelectorDropdown<br/>SafeSidebarContent<br/>Multiple inline components"]
    end
    
    subgraph "After: Modular Architecture"
        B1["EnhancedSidebar<br/>(index.tsx)"]
        B2["SidebarTopBar"]
        B3["Sidebar Variants<br/>Safe | Spaces"]
        B4["SafeSidebarWorkspaceHeader<br/>+ BackToSpaceButton<br/>+ NewTransactionButton"]
        B5["SpaceSelectorDropdown<br/>(refactored)"]
        B6["SafeSidebarContent"]
        B7["SidebarCommonFooter"]
        B8["SidebarSkeleton"]
        
        B1 --> B2
        B1 --> B3
        B3 --> B4
        B3 --> B5
        B3 --> B6
        B1 --> B7
        B1 --> B8
    end
    
    subgraph "New Hooks & Utilities"
        C1["useAddSafeToSpace<br/>- Space limit validation<br/>- Error handling"]
        C2["useCurrentSpaceId<br/>- Guard against falsy"]
    end
    
    B4 -.-> C1
    B5 -.-> C1
    B4 -.-> C2
    
    subgraph "UI Integration"
        D1["AddToSpacePopupModal<br/>+ Shadcn Dialog"]
        D2["Dialog Component<br/>from shadcn/ui"]
    end
    
    D1 --> D2
    
    style A1 fill:#ffcccc
    style B1 fill:#ccffcc
    style B2 fill:#ccffcc
    style B3 fill:#ccffcc
    style B4 fill:#ccffcc
    style B5 fill:#ccffcc
    style B6 fill:#ccffcc
    style B7 fill:#ccffcc
    style B8 fill:#ccffcc
    style C1 fill:#ccccff
    style C2 fill:#ccccff
    style D1 fill:#ffffcc
    style D2 fill:#ffffcc
```

## Checklist

- [x] I've tested the branch on mobile 📱
- [x] I've documented how it affects the analytics (if at all) 📊
  - Added tracking for "Open Space List" from space selector
  - Existing space creation tracking preserved
- [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻
  - 438+ new tests for sidebar components, hooks, and edge cases
  - Test coverage includes space limit validation, error handling, and UI states

---

## CLA signature

With the submission of this Pull Request, I confirm that I have read and agree to the terms of the [Contributor License Agreement](https://safe.global/cla).

### CODE REVIEW

**Comment by vmosharova** [comment]
@claude review

**Comment by vmosharova** [comment]
@claude review again

**Comment by clovisdasilvaneto** [comment]
@claude review

File: apps/web/src/components/ui/dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import * as React from 'react'
```
Comment (clovisdasilvaneto) [inline apps/web/src/components/ui/dialog.tsx:1]: I've added the same component 😆  we will have conflicts ahahahha

File: apps/web/src/features/spaces/components/Sidebar/hooks/useAddSafeToSpace.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+export const useAddSafeToSpace = ({ spaces, onSpaceAdded }: UseAddSafeToSpaceOptions): UseAddSafeToSpaceResult => {
+  const { safe } = useSafeInfo()
+  const chain = useCurrentChain()
+  const dispatch = useAppDispatch()
+  const [addSafeToSpace] = useSpaceSafesCreateV1Mutation()
+  const [loadingSpaceId, setLoadingSpaceId] = useState<number | null>(null)
+
+  const addToSpace = async (spaceId: number): Promise<boolean> => {
+    if (!chain?.chainId || !safe.address.value) return false
+    setLoadingSpaceId(spaceId)
```
Comment (clovisdasilvaneto) [inline apps/web/src/features/spaces/components/Sidebar/hooks/useAddSafeToSpace.ts:38]: why do we need this state? Can't we use { isLoading } from the `useSpaceSafesCreateV1Mutation` ?

File: apps/web/src/features/spaces/components/Sidebar/hooks/useAddSafeToSpace.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          message: 'Successfully added Safe to workspace.',
+          variant: 'success',
+          groupKey: 'add-safe-to-workspace-success',
+        }),
+      )
+      const space = spaces.find((s) => s.id === spaceId)
+      if (space) onSpaceAdded?.(space)
+      return true
+    } catch (error) {
+      console.error('Failed to add Safe to workspace.', error)
```
Comment (ChillYao) [inline apps/web/src/features/spaces/components/Sidebar/hooks/useAddSafeToSpace.ts]: Is this intended?

File: apps/web/src/features/spaces/components/Sidebar/variants/SafeSidebarVariant/SafeSidebarVariant.tsx [hunk 4]
```diff
@@ -10,83 +10,63 @@ import {
...
-        pathname: AppRoutes.spaces.index,
-        query: { spaceId },
-      })
-    }
-  }
+  const shouldRenderWorkspaceHeaderGroup =
+    workspaceHeader.variant === 'backToSpace' || !(isHydrated && isCounterfactualSafe)
 
   return (
     <SidebarContent className={css.sidebarContent}>
```
Comment (ChillYao) [inline apps/web/src/features/spaces/components/Sidebar/variants/SafeSidebarVariant/SafeSidebarVariant.tsx]: Seems for `css.sidebarContent`, `css.sidebarFooter` and `css.sidebarHeader`, I did not find them in our code base?

File: apps/web/src/features/spaces/components/Sidebar/SpacesEnhancedSidebar/SpacesEnhancedSidebar.tsx [hunk 5]
```diff
@@ -50,28 +54,41 @@ export const SpacesEnhancedSidebar = ({
...
+
+  const effectiveSelectedSpace = selectedSpace ?? addedToSpace ?? qualifiedSpace
+  const spaceName = effectiveSelectedSpace?.name ?? ''
 
   const sidebarType = isSpaceRoute ? 'spaces' : 'safe'
 
   return (
     <EnhancedSidebar
       type={sidebarType}
       spaceName={spaceName}
```
Comment (ChillYao) [inline apps/web/src/features/spaces/components/Sidebar/SpacesEnhancedSidebar/SpacesEnhancedSidebar.tsx]: Seems we don't need `spaceName` in child component anymore?

File: apps/web/src/components/ui/dialog.tsx (hunk 1 above)
Comment (vmosharova) [reply apps/web/src/components/ui/dialog.tsx:1]: I replaced mine with yours from the epic - conflicts are resolved now 👌🏻

File: apps/web/src/features/spaces/components/Sidebar/NewTransactionButton/SidebarActionButton.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    setTxFlow(<NewTxFlow />, undefined, false)
+    trackEvent(
+      { ...OVERVIEW_EVENTS.NEW_TRANSACTION, label: 'sidebar' },
+      { [MixpanelEventParams.SIDEBAR_ELEMENT]: 'New Transaction' },
+    )
+  }
+
+  if (isCounterfactualSafe) {
+    return <ActivateAccountButton />
+  }
```
Comment (clovisdasilvaneto) [inline apps/web/src/features/spaces/components/Sidebar/NewTransactionButton/SidebarActionButton.tsx:26]: I think this button should be outside the NewTransactionButton component, since it is related to ActivateAccount... Shall we change the name of the component then?

**Review by clovisdasilvaneto — APPROVED** [review]

File: apps/web/src/features/spaces/components/Sidebar/hooks/useAddSafeToSpace.ts (hunk 2 above)
Comment (vmosharova) [reply apps/web/src/features/spaces/components/Sidebar/hooks/useAddSafeToSpace.ts:38]: This allows to show a loading spinner right next to the specific Space where the Safe is added to. With the loader from the mutation hook we can only disable all items at once with an indication which one is loading. With `setLoadingSpaceId` we can do that:
image

File: apps/web/src/features/spaces/components/Sidebar/hooks/useAddSafeToSpace.ts (hunk 3 above)
Comment (vmosharova) [reply apps/web/src/features/spaces/components/Sidebar/hooks/useAddSafeToSpace.ts]: You are right, it makes sense to show an error in the toaster, not only in the console. Fixed: [51cce42](https://github.com/safe-global/safe-wallet-monorepo/pull/7602/commits/51cce42cabc49db0f575252d3a73d16a84b11e58)

File: apps/web/src/features/spaces/components/Sidebar/variants/SafeSidebarVariant/SafeSidebarVariant.tsx (hunk 4 above)
Comment (vmosharova) [reply apps/web/src/features/spaces/components/Sidebar/variants/SafeSidebarVariant/SafeSidebarVariant.tsx]: Good catch, thank you! Removed: [da2e62c](https://github.com/safe-global/safe-wallet-monorepo/pull/7602/commits/da2e62cd3c2b9a686ebe0639ddee315566175dce)

File: apps/web/src/features/spaces/components/Sidebar/SpacesEnhancedSidebar/SpacesEnhancedSidebar.tsx (hunk 5 above)
Comment (vmosharova) [reply apps/web/src/features/spaces/components/Sidebar/SpacesEnhancedSidebar/SpacesEnhancedSidebar.tsx]: We still need it as it is passed as props down to SpacesSidebarContent which uses it to render the space name in the header/avatar. But you are right, it can be optimized, refactored here: [e8c4960](https://github.com/safe-global/safe-wallet-monorepo/pull/7602/commits/e8c4960d8a5b979fb9610d696031ee07b4b67fef)

File: apps/web/src/features/spaces/components/Sidebar/NewTransactionButton/SidebarActionButton.tsx (hunk 6 above)
Comment (vmosharova) [reply apps/web/src/features/spaces/components/Sidebar/NewTransactionButton/SidebarActionButton.tsx:26]: I'd keep it here. They occupy the same UI slot and the conditional is straightforward.

But you are right about the component name, it should be clearer. Renamed `NewTransactionButton` to `SidebarActionButton` (also to avoid confusing with the old MUI `NewTxButton`): [1801502](https://github.com/safe-global/safe-wallet-monorepo/pull/7602/commits/18015021442a056c75795e732724bf5092d55595)
